packages/components/addon/components/hds/dialog-primitive/header/index.hbs

```diff
@@ -0,0 +1,36 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+<div class="hds-dialog-primitive__header" ...attributes>
+  {{#if @icon}}
+    <FlightIcon class="hds-dialog-primitive__icon" @name={{@icon}} @size="24" @isInlineBlock={{false}} />
+  {{/if}}
+
+  <Hds::Text::Display class="hds-dialog-primitive__title" @tag="div" @size="300" @weight="semibold" id={{@id}}>
+    {{#if @tagline}}
+      <Hds::Text::Body class="hds-dialog-primitive__tagline" @tag="div" @size="100" @weight="regular">
+        {{@tagline}}
+      </Hds::Text::Body>
+    {{/if}}
+
+    {{yield}}
+  </Hds::Text::Display>
+
+  <Hds::DismissButton class="hds-dialog-primitive__dismiss" {{on "click" this.onDismiss}} />
+</div>
+
+{{!-- DIFFERENCES:
+* FLYOUT HEADER TAGLINE - uses @color="faint"
+
+* FLYOUT TAGGLINE:
+<Hds::Text::Body class="hds-flyout__tagline" @tag="div" @size="100" @weight="regular" @color="faint">
+  {{@tagline}}
+</Hds::Text::Body>
+
+* MODAL TAGGLINE:
+<Hds::Text::Body class="hds-modal__tagline" @tag="div" @size="100" @weight="regular">
+  {{@tagline}}
+</Hds::Text::Body>
+--}}
```

Comment:
This is just here as a temporary reference while I work on this. It will be deleted in the end.